### PR TITLE
Add post on database prototyping

### PR DESCRIPTION
**Pull request by peteryates**
This post (hopefully) provides an overview of why we decided to start ECF2 by tackling the biggest problem first, and how doing this work up front is easier than trying to do it while the application is being developed.

- **Begin work on database first design post**
- **Add sections on rebuilding and prototypes**

### CODE REVIEW

File: app/posts/ecf-v2/initial-release-of-ecf-2.md
Comment (peteryates) [inline app/posts/ecf-v2/initial-release-of-ecf-2.md:1]: The whitespace deletion here was applied by my editor when saving, the only actual change is adding a `v` to the image path as the directory was renamed for consistency.

File: app/posts/ecf-v2/2024-07-15-designing-the-database-first.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+To complicate matters further, only relevant parties can see parts of the induction record that belong to them.
+
+### Finding a better way
+
+We first started seriously thinking about how to better record training and induction data in August 2023 and held a workshop where we [came up with several schema ideas](https://github.com/dfE-Digital/ecf-data-model-prototypes) and discussed them at length. Later we [whittled them down to one](https://github.com/DFE-Digital/ecf-data-model-prototypes-separate/), a model which stored each kind of data separately.
+
+Holding each kind of data in its own table has several advantages:
+
+1. when something changes we only need to update the corresponding table
+2. more of the model is routed in reality, we need to introduce fewer abstract concepts which are often a point of confusion
```
Comment (claire-hughez) [inline app/posts/ecf-v2/2024-07-15-designing-the-database-first.md:57]:
```suggestion
2. more of the model is routed in reality
```

Comment (claire-hughez) [reply app/posts/ecf-v2/2024-07-15-designing-the-database-first.md:57]: the second part of this bullet does not make sense to me, is it necessary?

File: app/posts/ecf-v2/2024-07-15-designing-the-database-first.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+With this schema we'd have to add extra validation logic to cope with these requirements. Splitting tenureships into two tables, one for ECTs and one for mentors, was much cleaner.
+
+### The current situation
+
+![ECF2 - now with separate ECT and mentor at school periods](/ecf-v2/designing-the-database-first/ecf-2-induction-data.png)
+
+Now we have a model we're fairly confident in. It's coped well with the [scenarios we've thrown at it](https://github.com/DFE-Digital/ecf2-rails-prototype/pull/2) and has been scrutinised by lots of people across the CPD programme.
+
+It's definitely not done yet though, and we'll continue to refine it as the application is developed. Doing these many iterations now, before we have an application and data to migration, has been a rewarding exercise and given us confidence that when we begin the actual build we're setting off in the right direction.
```
Comment (claire-hughez) [inline app/posts/ecf-v2/2024-07-15-designing-the-database-first.md:76]: Can we add a line on testing this with lead providers or other external users?

**Review by claire-hughez — COMMENTED** [review]
Hey Pete,

Looks really good to me, just a few points. I think we could go into a bit more detail on the problems in the current data model but we don't have to.

**Pull request merged**